### PR TITLE
Feature/gh build docs

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
What is the problem this work solves, including
We want to be able to automatically update the documentation when branches merged into main  

Solution
========
What I/we did to solve this problem
- added a workflow in github actions  
- revised the makefile command 
- addressed minor styling issues in `docs/HandoffSpecification.adoc` to eliminate warnings

(there are some remainning warning messages from `asciidoctor` about duplicate IDs, seemed renaming IDs in `DatabaseSchema.adoc` or `DatasetHandoffRef.adoc` is not the correct way to fix them, how do we modify the AsciiDoc files to fix the warnings?)

with @meganrm 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. check if the build_docs workflow does update documentation when branches merged into main

### CODE REVIEW

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Docuemntation
```
Comment (toloudis) [inline .github/workflows/build-docs.yml:1]:
```suggestion
name: Documentation
```

Comment (meganrm) [reply .github/workflows/build-docs.yml:1]: I don't think I ever would have noticed this :)

**Review by meganrm — APPROVED** [review]
Approve after fixing the typo Dan pointed out

File: .github/workflows/build-docs.yml (hunk 1 above)
Comment (rugeli) [reply .github/workflows/build-docs.yml:1]: ah! Thank you!!

**Pull request merged**